code/modules/reagents/reagent_containers/food/lunch.dm

```diff
@@ -107,11 +107,11 @@ var/global/list/lunchables_ethanol_reagents_ = list(
 		.[initial(O.name)] = lunch
 	return sortTim(., /proc/cmp_text_asc)
 
-/proc/init_lunchable_reagent_list(var/list/banned_reagents, var/reagent_types)
+/proc/init_lunchable_reagent_list(var/list/banned_reagents, var/reagent_type)
 	. = list()
-	for(var/reagent_type in subtypesof(reagent_types))
-		if(reagent_type in banned_reagents)
+	for(var/reagent_subtype in decls_repository.get_decls_of_subtype(reagent_type))
```

Comment:
the `get_decls_of_subtype` changes for ethanol aren't actually necessary because ethanol is an abstract type anyway. in fact this might cause issues if you pass in a non-abstract type to this proc ever. technically it's not a big deal but eh